server/src/main/java/org/apache/iotdb/db/qp/constant/DatetimeUtils.java

```diff
@@ -475,6 +475,66 @@ public static long convertDatetimeStrToLong(String str, ZoneOffset offset, int d
     return getInstantWithPrecision(str, timestampPrecision);
   }
 
+  /**
+   * convert duration string to millisecond, microsecond or nanosecond.
+   */
+  public static long convertDurationStrToLong(long value, String unit, String timestampPrecision) {
+
+    long res = value;
+    switch (unit) {
+      case "y":
+        res *= 365 * 86400_000;
+        break;
+      case "mo":
+        res *= 30 * 86400_000;
+        break;
+      case "w":
+        res *= 7 * 86400_000;
+        break;
+      case "d":
+        res *= 86400_000;
+        break;
+      case "h":
+        res *= 3600_000;
+        break;
+      case "m":
+        res *= 60_000;
+        break;
+      case "s":
+        res *= 1_000;
+        break;
+      default:
```

Comment:
Where is "ns"?

Plus, would it be better to make `String unit` a constant or enum or something?

Reply:
Plus Plus, why doesn't the default branch throw an exception?

Reply:
Plus Plus Plus, if this miss of "ns" is a bug, better add tests to cover this.

Reply:
> Plus Plus, why doesn't the default branch throw an exception?
> Plus Plus Plus, if this miss of "ns" is a bug, better add tests to cover this.
This function has the second half.
```
if (timestampPrecision.equals("us")) {
      if (unit.equals("ns")) {
        return value / 1000;
      } else if (unit.equals("us")) {
        return value;
      } else {
        return res * 1000;
      }
    } else if (timestampPrecision.equals("ns")) {
      if (unit.equals("ns")) {
        return value;
      } else if (unit.equals("us")) {
        return value * 1000;
      } else {
        return res * 1000_000;
      }
    } else {
      if (unit.equals("ns")) {
        return value / 1000_0000;
      } else if (unit.equals("us")) {
        return value / 1000;
      } else {
        return res;
      }
    }
```

Reply:
Yes, I will make String unit a  enum.